### DISCUSSION TITLE
bitbucket-server producing error in dry-run with non-onboarded repos

### DISCUSSION THREAD

**Discussion by ps-jay** [renovatebot/renovate#21692] [Suggest an Idea]
**What Renovate type, platform and version are you using?**

Self-hosted Renovate, using the `22.25.6-slim` image, scanning bitbucket-server.

**Describe the bug**

Throwing an error when a repo is not onboarded, and does not have an onboarding branch

```
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "1CS1BCOd5",
           "repository": "kpe/springboot-jdbc-connector-event",
           "err": {
             "message": "Cannot check modification for branch that does not exist: renovate/configure",
             "stack": "Error: Cannot check modification for branch that does not exist: renovate/configure\n    at Object.isBranchModified (/usr/src/app/node_modules/renovate/lib/util/git/index.ts:430:11)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n    at Object.ensureOnboardingPr (/usr/src/app/node_modules/renovate/lib/workers/repository/onboarding/pr/index.ts:68:7)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:41:5)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:66:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:10:22"
           },
           "msg": "Repository has unknown error"
         }
       ]
```

**Relevant debug logs**

```
DEBUG: config.repoIsOnboarded=false (repository=kpe/springboot-jdbc-connector-event)
DEBUG: ensureOnboardingPr() (repository=kpe/springboot-jdbc-connector-event)
DEBUG: getBranchPr(renovate/configure) (repository=kpe/springboot-jdbc-connector-event)
DEBUG: findPr(renovate/configure, "undefined", "open") (repository=kpe/springboot-jdbc-connector-event)
DEBUG: getPrList() (repository=kpe/springboot-jdbc-connector-event)
DEBUG: returning cached PR list (repository=kpe/springboot-jdbc-connector-event)
DEBUG: DID NOT Found PR from branch #renovate/configure (repository=kpe/springboot-jdbc-connector-event)
DEBUG: Filling in onboarding PR template (repository=kpe/springboot-jdbc-connector-event)
ERROR: Repository has unknown error (repository=kpe/springboot-jdbc-connector-event)
       "err": {
         "message": "Cannot check modification for branch that does not exist: renovate/configure",
         "stack": "Error: Cannot check modification for branch that does not exist: renovate/configure\n    at Object.isBranchModified (/usr/src/app/node_modules/renovate/lib/util/git/index.ts:430:11)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n    at Object.ensureOnboardingPr (/usr/src/app/node_modules/renovate/lib/workers/repository/onboarding/pr/index.ts:68:7)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:41:5)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:66:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:10:22"
       }
```

**To Reproduce**

Sorry, our bitbucket-server instance is "behind the firewall".

**Additional context**

Seems to be related to the recent changes to error handling in the bitbucket-server module maybe?

We were experiencing a similar looking issue on empty repositories in v22.23.0, but that has since been resolved by v22.25.6 - presumably by #7070 ?  Empty repos were also handled okay in v22.17.2.

**Comment by viceice** [comment]
I've seen them too on repos, which are not onboarded (have no renovate config)

**Comment by ps-jay** [comment]
Further testing: it only occurs in `dry-run` mode

**Comment by viceice** [comment]
That why i'm seeing it very rarely, as most of our repo's are not onboarded a very short time and dry-run mode is used for renovate self updating.

**Comment by ps-jay** [comment]
Yeah.  So it's probably not high priority.

I don't observe the issue in v21.33.15, but I do in v22.0.7.

**Comment by viceice** [comment]
I think it's due to big internal changes. We are splitting git realeted code from platform code, so it looks like a small regression.

A workaround is to simply onboard the repo, you can disable renovate if the repo should be ignored, or you can use the auto discover filter to ignore.

**Comment by ps-jay** [comment]
Yup, no worries 👍 

Feel free to close this issue if from your perspective it is not worth worrying about.  I'm not concerned anymore.

---

My use-case is that I want to dry-run our entire bitbucket-server, to gather metrics on Renovate's uptake.

I end up parsing the logs for strings such as `Repo is onboarded`, `Repository has no package files`, `Repository is empty`, to end up with the following metrics:
* number of repos onboard
* number of repos not onboard
* number of repos that are not renovatable

It'd be cool if there were a special `dry-run` mode that did not actually scan for package updates, just for this purpose.

**Comment by ahippler** [comment]
Please don't close the issues. Its a very annoying bug.
